docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+
+As long as the selected first few leaders are honest, faithful, and even good,
+the network will be not only **live**, but also **fair**. (note that it's always **safe** regardless of the leader's behavior; safety is guaranteed by the BFT consensus)
+
+### So What?
+
+To summarize, the leader has **responsibilities** and **authorities**.
+Actually, this is not so different from other blockchains that uses a leader-based BFT consensus. The leader should be online for the whole round, to receive enough amount of transactions (in the 'mempool'), and the leader chooses which transactions to include in the block. The only difference is that we don't have a user-signed transaction but a governance-signed agenda instead, and that we have an additional protocol for the chat finalization.
+
+So how the 'other blockchains' deal with the leader's responsibilities and authorities? It's simple; they have a very short round time (several seconds),
+and rotate the round order every block, so that the actual experience of 'users' (corresponding to 'members' in Simperby) converges to the average of the validator population. In other words, even if some block proposer is lazy (not faithful) or bad (attempting censorship), the users will not severely suffer from it, because it doesn't take long until the next block proposer comes to serve their transactions.
+
+This kind of problem can be characterized as a **single point of failure** problem, that we particularly suffer from because of the long-round and stable-leader conditions.
+
+The only ways to resolve this are:
+
+1. The governance approves an agenda of impeaching the leader, that is, removing it from the validator list. This is an ideal solution because it instantly changes the leader.
+2. If the leader doesn't include such agenda in their block, there is the last resort: the **governance doesn't approve any agenda**, so that the proposer can't propose any valid block. Remind that an empty block is invalid. This will escape the round after the long round timeout.
+
+## Consensus
+
+So far, we have discussed about how the governance and human communication of Simperby work, assuming an abstract underlying BFT consensus. It works well, but we found that there is a serious problem of time lagging in case of an laziness. There coule be even permanent censorship in case of a bad leader, which can be still resolved after a long time of round if the governance doesn't approve any agenda. This works, but it's not really efficient considering that 'being long' can be a week or more.
+
+Now we consider the consensus layer as an option for dealing with this, but unfortunately, **there is no way to resolve it in the consensus layer, ideally**. Unlike Byzantine or 'dishonest' behaviaor which can be directly verified by some cryptographic evidence (e.g. double pre-votes), 'unfaithful' or 'bad' leader can not be handled by the consensus because it's not a violoation of the consensus protocol.
```

Comment:
```suggestion
Now we consider the consensus layer as an option for dealing with this, but unfortunately, **there is no way to resolve it in the consensus layer, ideally**. Unlike Byzantine or 'dishonest' behavior which can be directly verified by some cryptographic evidence (e.g. double pre-votes), 'unfaithful' or 'bad' leader can not be handled by the consensus because it's not a violation of the consensus protocol.
```